### ISSUE TITLE
zld performance on macos

### ISSUE THREAD

**Issue by randomNinja1000** [kubkon/zld#138]
Hello,

I have been experimenting with zig lately and I believe I am seeing some unexpected time / memory consumption from zld.

TL;DR is on an example I post here I see 10x slower link times with zld https://github.com/randomNinja1000/zld-memory-issues and very high memory consumptions.

In other scenarios, this easily blows up my 64GB mem + 64 GB swap when linking multiple larger objects in parallel.

Is this a known issue?

**Comment by motiejus** [comment]
Not known. Cc @kubkon

**Comment by motiejus** [comment]
```
motiejus@bookworm:~/code/zld-memory-issues/objs$ git lfs pull
batch response: This repository is over its data quota. Account responsible for LFS bandwidth should purchase more data packs to restore access.                                                                    
Failed to fetch some objects from 'https://github.com//randomNinja1000/zld-memory-issues.git/info/lfs'
```

Is there a way for you to put them separately to a tarball?

**Comment by kubkon** [comment]
Fix in #140 for standalone `zld`, and in a branch for upstream `zig` in https://github.com/ziglang/zig/commit/ed7073c6300d6ae3ec4b370c1e13c2bbf11bc08c
